pkg/yurtmanager/controller/autonomy/autonomy_controller.go

```diff
@@ -0,0 +1,135 @@
+	"github.com/openyurtio/openyurt/pkg/projectinfo"
+)
+
+var (
+	controllerResource = v1.SchemeGroupVersion.WithKind("Node")
+)
+
+type ReconcileAutonomy struct {
+	client.Client
+}
+
+func Format(format string, args ...interface{}) string {
+	s := fmt.Sprintf(format, args...)
+	return fmt.Sprintf("%s: %s", names.AutonomyController, s)
+}
+
+func Add(ctx context.Context, c *appconfig.CompletedConfig, mgr manager.Manager) error {
+	klog.Infof(Format("autonomy-controller add controller %s", controllerResource.String()))
+	return add(mgr, c, newReconciler(c, mgr))
+}
+
+func newReconciler(_ *appconfig.CompletedConfig, mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileAutonomy{Client: mgr.GetClient()}
+}
+
+func add(mgr manager.Manager, cfg *appconfig.CompletedConfig, r reconcile.Reconciler) error {
+	c, err := controller.New(names.AutonomyController, mgr, controller.Options{
+		Reconciler:              r,
+		MaxConcurrentReconciles: int(cfg.Config.ComponentConfig.AutonomyController.ConcurrentAutonomyWorkers),
+	})
+	if err != nil {
+		return err
+	}
+
+	nodePredicate, nodeMapFunc := prepare()
+	err = c.Watch(&source.Kind{Type: &v1.Node{}}, handler.EnqueueRequestsFromMapFunc(nodeMapFunc), nodePredicate)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *ReconcileAutonomy) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	klog.V(2).Infof("Reconcile Node %s/%s Start.", req.Namespace, req.Name)
+	var node v1.Node
+	err := r.Client.Get(ctx, req.NamespacedName, &node)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == appsv1beta1.NodeAutonomy && condition.Status == v1.ConditionTrue {
+			if node.Labels == nil {
+				node.Labels = make(map[string]string)
+			}
+			node.Labels[projectinfo.GetAutonomyStatusLabel()] = "true"
+		}
+	}
+	if err := r.Client.Update(ctx, &node); err != nil {
+		klog.Errorf("failed to update node %s/%s, err: %s", node.Namespace, node.Name, err)
+		return reconcile.Result{}, err
+	}
+	return reconcile.Result{}, nil
+}
+
+func prepare() (predicate.Funcs, func(node client.Object) (req []reconcile.Request)) {
+	nodePredicate := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			return true
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			oldNode := e.ObjectOld.(*v1.Node)
+			newNode := e.ObjectNew.(*v1.Node)
+			if reflect.DeepEqual(oldNode.Annotations, newNode.Annotations) {
+				return false
+			}
+			return true
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+		GenericFunc: func(genericEvent event.GenericEvent) bool {
+			return false
+		},
+	}
+	nodeMapFunc := func(node client.Object) (req []reconcile.Request) {
+		labels := node.GetLabels()
+		annotations := node.GetAnnotations()
+		if labels[projectinfo.GetEdgeWorkerLabelKey()] == "true" && annotations[projectinfo.GetAutonomyAnnotation()] == "true" {
```

Comment:
these `if condition` should be removed into nodePredicate

Reply:
okay